### PR TITLE
[CIRC-1895] Improve logging circulation resources: E - N

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose

Improve logging resource package from EndPatronActionSessionResource to NoticeCirculationRulesEngineResource in alphabetic order

Resolves: [CIRC-1895](https://issues.folio.org/browse/CIRC-1895)

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/ExpiredSessionProcessingResource.java [hunk 1]
```diff
@@ -74,6 +78,7 @@ private void process(RoutingContext routingContext) {
   }
 
   private CompletableFuture<Result<ZonedDateTime>> defineExpiredTime(Integer timeout) {
+    log.debug("defineExpiredTime:: parameters timeout: {}", timeout);
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/ExpiredSessionProcessingResource.java:81]: What about process() method? We logged it in ExpiredActualCostProcessingResource.java

File: src/main/java/org/folio/circulation/resources/ExpiredSessionProcessingResource.java [hunk 2]
```diff
@@ -88,6 +93,7 @@ private CompletableFuture<Result<Void>> attemptEndSessions(
       .collect(Collectors.toList());
 
     if (existingExpiredSessions.isEmpty()) {
+      log.info("attemptEndSessions:: no existing expired sessions");
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/ExpiredSessionProcessingResource.java:96]: Should we log the beginning of the method too?

File: src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java [hunk 3]
```diff
@@ -52,4 +59,13 @@ protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> handleNoti
       .handleNotices(noticesResult.getRecords())
       .thenApply(mapResult(v -> noticesResult));
   }
+
+  private Result<MultipleRecords<ScheduledNotice>> logNotices(
+    MultipleRecords<ScheduledNotice> records) {
+
+    log.info("logNotices:: found notices: {}",
+      () -> LogUtil.collectionAsString(records.getRecords()));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java:67]: multipleRecordsAsString()

File: src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java [hunk 4]
```diff
@@ -38,7 +44,8 @@ protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> findNotice
     return scheduledNoticesRepository.findNotices(
       ClockUtil.getZonedDateTime(), true,
       List.of(DUE_DATE, AGED_TO_LOST),
-      CqlSortBy.ascending("nextRunTime"), pageLimit);
+      CqlSortBy.ascending("nextRunTime"), pageLimit)
+      .thenApply(r -> r.next(this::logNotices));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java:48]: Is this indentation correct?

File: src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java [hunk 5]
```diff
@@ -52,4 +59,13 @@ protected CompletableFuture<Result<MultipleRecords<ScheduledNotice>>> handleNoti
       .handleNotices(noticesResult.getRecords())
       .thenApply(mapResult(v -> noticesResult));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/LoanScheduledNoticeProcessingResource.java:60]: Should we log this method?

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 6]
```diff
@@ -77,6 +83,7 @@ private CompletableFuture<Result<Void>> sendLoanNotice(Loan loan, NoticeEventTyp
   }
 
   private Result<Loan> validateLoan(Loan loan) {
+    log.debug("validateLoan:: parameters loan: {}", loan);
     List<ValidationError> errors = new ArrayList<>();
 
     if (loan == null) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:89]: Why are we skipping these ifs?

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 7]
```diff
@@ -111,6 +118,7 @@ private Result<Loan> publishNoticeErrorEvent(HttpFailure failure, Loan loan,
 
 
   private CompletableFuture<Result<Void>> sendNotice(Loan loan, NoticeEventType eventType) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:120]: Why have we skipped publishNoticeErrorEvent()?

File: src/main/java/org/folio/circulation/resources/EndPatronActionSessionResource.java [hunk 8]
```diff
@@ -48,13 +53,15 @@ private void process(RoutingContext routingContext) {
...
       } else {
         EndPatronSessionRequest endSessionRequest = result.value();
-        patronActionSessionService.endSessions(
-          endSessionRequest.getPatronId(),
-          endSessionRequest.getActionType());
-
+        String patronId = endSessionRequest.getPatronId();
+        PatronActionType actionType = endSessionRequest.getActionType();
+        patronActionSessionService.endSessions(patronId, actionType);
+        log.info("process:: session ended successfully: getPatronId(): {}, actionType: {}",
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/EndPatronActionSessionResource.java:63]:
```suggestion
        log.info("process:: session ended successfully: patronId(): {}, actionType: {}",
```

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 9]
```diff
@@ -77,15 +83,19 @@ private CompletableFuture<Result<Void>> sendLoanNotice(Loan loan, NoticeEventTyp
   }
 
   private Result<Loan> validateLoan(Loan loan) {
+    log.debug("validateLoan:: parameters loan: {}", loan);
     List<ValidationError> errors = new ArrayList<>();
 
     if (loan == null) {
+      log.info("validateLoan:: loan in null");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:90]:
```suggestion
      log.info("validateLoan:: loan is null");
```

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 10]
```diff
@@ -77,15 +83,19 @@ private CompletableFuture<Result<Void>> sendLoanNotice(Loan loan, NoticeEventTyp
...
       errors.add(new ValidationError("Loan is null", "loan", null));
     } else {
       if (loan.getUser() == null) {
+        log.info("validateLoan:: user in null");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:94]:
```suggestion
        log.info("validateLoan:: user is null");
```

File: src/main/java/org/folio/circulation/resources/LoanNoticeSender.java [hunk 11]
```diff
@@ -77,15 +83,19 @@ private CompletableFuture<Result<Void>> sendLoanNotice(Loan loan, NoticeEventTyp
...
         errors.add(new ValidationError("User is null", "user", null));
       }
       if (loan.getItem() == null || loan.getItem().isNotFound()) {
+        log.info("validateLoan:: user in null");
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/resources/LoanNoticeSender.java:98]:
```suggestion
        log.info("validateLoan:: user is null");
```

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**